File: packtrain-policy-service/src/data/GradingStartDTO.ts
export class GlobalAssignmentMetadata {
    assignmentId!: string;
    canvasMaxScore!: number;
    canvasMinScore!: number;
    externalMaxScore!: number;
    initialDueDate!: Date;
}

export default class GradingStartDTO {
    migrationId!: string;
    policyURI!: string;
    scoreCreatedRoutingKey!: string;
    rawGradeRoutingKey!: string;

    globalMetadata!: GlobalAssignmentMetadata;
}

File: packtrain-service/src/main/java/edu/mines/packtrain/data/policyServer/GradingStartDTO.java
package edu.mines.packtrain.data.policyServer;

import java.net.URI;
import java.time.Instant;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
public class GradingStartDTO {
    @AllArgsConstructor
    @Data
    public static class GlobalAssignmentMetadata {
        private UUID assignmentId;
        private double canvasMaxScore;
        private double canvasMinScore;
        private double externalMaxScore;
        private String initialDueDate;
    }

    private UUID migrationId;
    private URI policyURI;
    private String scoreCreatedRoutingKey;
    private String rawGradeRoutingKey;

    private GlobalAssignmentMetadata globalMetadata;

}
